src/arch/xtensa/include/arch/timer.h

```diff
@@ -39,6 +39,7 @@
 struct timer {
 	uint32_t id;
 	uint32_t irq;
+	unsigned int cpu;
```

Comment:
uint32_t

Reply:
any specific reason for this? I'm pretty sure we aren't going to overflow the smallest imaginable "int" with the number of our cores for a rather long time ;-)

Reply:
uhm, sorry, I disagree... First of all the above void* is not a fixed-width field. Then "irq" to this moment is a bit-mask, so, I can understand why it's fixed-width. ID actually doesn't make sense to be fixed-width to me, but it'll be gone in a couple of patches. Further, I don't think we're going to deal with systems / compilers, on which int isn't 32-bit? But in general, I think the rule in the kernel is, that fixed-width types should *only* be used where required, e.g. when mimicking hardware registers or for similar uses. Other variables should be typed according to their context, meaning - int, long, char, unsigned int etc.